### PR TITLE
CB-5938 Empty tag value breaks environment creation

### PR DESCRIPTION
**Pull request by doktoric**
unit test added

### CODE REVIEW

File: template-manager-tag/src/main/java/com/sequenceiq/cloudbreak/tag/DefaultCostTaggingService.java [hunk 1]
```diff
@@ -57,14 +57,24 @@
     public Map<String, String> mergeTags(CDPTagMergeRequest request) {
         LOGGER.debug("About to merge tag(s)...");
         Map<String, String> result = new HashMap<>();
-        result.putAll(request.getRequestTags());
-        for (String key : request.getEnvironmentTags().keySet()) {
-            addTagIfNotPresented(result, request, key);
+        for (Map.Entry<String, String> entry : request.getRequestTags().entrySet()) {
```
Comment (keyki) [inline template-manager-tag/src/main/java/com/sequenceiq/cloudbreak/tag/DefaultCostTaggingService.java]: request.getRequestTags() can be null

File: template-manager-tag/src/main/java/com/sequenceiq/cloudbreak/tag/DefaultCostTaggingService.java [hunk 2]
```diff
@@ -57,14 +57,24 @@
...
+            if (!keyOrValueIsEmpty(entry.getKey(), entry.getValue())) {
+                addTagIfNotPresented(result, request, entry.getKey());
+            }
         }
         LOGGER.debug("The following requested tag(s) will be applied prepared: {}", result);
         return result;
     }
 
+    private boolean keyOrValueIsEmpty(String key, String value) {
+        return (key == null || key.isBlank() || key.isEmpty()) || (value == null || value.isBlank() || value.isEmpty());
```
Comment (keyki) [inline template-manager-tag/src/main/java/com/sequenceiq/cloudbreak/tag/DefaultCostTaggingService.java]: do we need isBlank and isEmpty?

Comment (doktoric) [reply template-manager-tag/src/main/java/com/sequenceiq/cloudbreak/tag/DefaultCostTaggingService.java]: yes

File: template-manager-tag/src/main/java/com/sequenceiq/cloudbreak/tag/DefaultCostTaggingService.java (hunk 1 above)
Comment (doktoric) [reply template-manager-tag/src/main/java/com/sequenceiq/cloudbreak/tag/DefaultCostTaggingService.java]: it can not!

**Pull request merged**